### DISCUSSION TITLE
NXP LPC55S36 Zephyr Port

### DISCUSSION THREAD

**Discussion by Phase777** [zephyrproject-rtos/zephyr#46421] [Q&A] [answered]
I am working on a Zephyr port for the LPC55S36JHI48 and need to create the LPC55S36JHI48-pinctrl.h file.  I am referencing the existing LPC55S06JHI48-pinctrl.h(attached below) to create LPC55S36JHI48-pinctrl.h.  At the top of LPC55S06JHI48-pinctrl.h there is a comment that says "NOTE: File generated by lpc_cfg_utils.py from LPC55S06JHI48/signal_configuration.xml".  Is it possible to use lpc_cfg_utils.py to generate LPC55S36JHI48-pinctrl.h?  If so where do I find lpc_cfg_utils.py?

 

Any other suggestions or tools for doing this Zephyr port would be helpful. Thanks.
[LPC55S06JHI48-pinctrl.zip](https://github.com/zephyrproject-rtos/zephyr/files/8873090/LPC55S06JHI48-pinctrl.zip)

**Comment by OUNAVCON** [comment]
@danieldegrasse,

What can be done to help this along? There is a duplicate issue on the Community forum as well.
https://community.nxp.com/t5/LPC-Microcontrollers/LPC55S36-Zephyr-Port/m-p/1471569#M49118

**Comment by danieldegrasse** [comment, marked as answer]
Hi @Phase777- The lpc_cfg_utils.py script has not been merged into NXP's HAL yet, but it is available here: https://github.com/zephyrproject-rtos/hal_nxp/pull/159. The script can generate pin control headers using the configuration tool data for an SOC. To do so, take the following steps:
1. start by downloading version 11 of the [MCUXpresso config tools](https://www.nxp.com/design/software/development-software/mcuxpresso-software-and-tools-/mcuxpresso-config-tools-pins-clocks-peripherals:MCUXpresso-Config-Tools). 
2. Once you have that tool downloaded, attempt to create a new pin/clock configuration file for the LPC55S36- this will cause the configuration tools to download the required processor collateral. 
3. Once that processor data has been downloaded, you need to locate it on your filesystem. On a Windows machine, this is at `C:\ProgramData\NXP\mcu_data_v11\processors`. There should be a folder here named "LPC55S36" with your processor data
4. With this processor data folder located, you can run the generation script to create the pin control file. Check out that PR with the lpc configuration scripts, and run the script with arguments like the following:
`.\mcux\scripts\lpc_gen_headers.py "C:\ProgramData\NXP\mcu_data_v11\processors\LPC55S36"` (note that the path to the processors folder here may vary, and should be taken from step 3).
5. This command should output the file `LPC55S36JHI48-pinctrl.h`

Please feel free to reach out if the above steps are unclear- I can also provide the generated file directly via a PR, if that is preferred.

**Reply by OUNAVCON** [reply]
@danieldegrasse,

Thank you for this information. It is simplifies what needs to be done to create Zephyr ports for LPC processors. NXPs contribution to the Zephyr project is one of the reasons that this combination of Silicon Vendor and Open OS have such powerful value propositions. It's two great individual organizations coming together to great something unique and powerful. This level of enablement is whey I continue to choose NXP.

**Reply by Phase777** [reply]
@danieldegrasse 

Thanks for the help! I was able to generate LPC55S36JHI48-pinctrl.h.  It looks like if I run the other lpc scripts in the pull request I can generate all the required files for the zephyr port.  Is this correct?  If so can you provide some instruction on running them?  Thanks!

**Reply by danieldegrasse** [reply]
`lpc_gen_dts.py` can be used to generate the pin control file only- the soc level DTSI still needs to be handwritten. To generate the pin control file, you can populate your MCUXpresso Config tools MEX configuration file with pin mux settings, and then run that script. The invocation will look similar to the following:

`.\mcux\scripts\lpc_gen_headers.py --cfg-tool-root "C:\ProgramData\NXP\mcu_data_v11\processors" PATH_TO_MEX_FILE`

That should result in a `pinctrl.dtsi` file for your board.

**Comment by Navin-Sankar** [comment]
Hi @danieldegrasse , how to generate the pin control file by SDK alone without using MCUXpresso Config tools.?
Currently I am looking for a pin control support for [lpcxpresso51u68](https://github.com/zephyrproject-rtos/zephyr/pull/37148) board.

**Reply by danieldegrasse** [reply]
@Navin-Sankar You will need to use MCUXpresso Config tools, as this is the data format that the scripts understand. You should be able to follow the steps given above to generate these files, provided the LPC51u68 pin control peripheral is similar to one that is currently supported.
